src/osd/PrimaryLogPG.cc

```diff
@@ -584,6 +585,8 @@ bool PrimaryLogPG::should_send_op(
   }
   if (async_recovery_targets.count(peer) && peer_missing[peer].is_missing(hoid)) {
     should_send = false;
+    missing_loc.add_location(hoid, peer);
```

Comment:
missing_loc is the location of valid copies of missing objects - should add the acting set where the object is not missing